Vector/Vector.hpp

```diff
@@ -0,0 +1,294 @@
+
+  void push(T value) { Vector<T>::push_back(value); };
+  void top() { Vector<T>::back();};
+  void pop() { Vector<T>::pop_back();};
+
+private:
+};
+
+template <class T>
+Vector<T>::Vector(size_t size)
+    : data_{new T[size]{}}, capacity_{size * delta}, size_{size} {}
+
+template <class T>
+Vector<T>::Vector(const Vector &rhs)
+    : data_{new T[rhs.size_]{}}, capacity_{rhs.capacity_}, size_{rhs.size_} {}
+
+template <class T>
+Vector<T>::Vector(std::initializer_list<T> args)
+    : capacity_{args.size() * delta} {
+  data_ = new T[args.size()];
+  size_ = 0;
+  for (T x : args) {
+    data_[size_++] = x;
+  }
+}
+
+template <class T> Vector<T>::Vector(const T *begin, const T *const end) {
+  if (begin == nullptr || end == nullptr || begin == end) {
+    return;
+  }
+  size_ = std::distance(begin, end);
+  capacity_ = size_ * delta;
+  data_ = new int[size_];
+  for (size_t i = 0; i < size_; ++i) {
+    data_[i] = *begin++;
+  }
+}
+
+template <class T>
+Vector<T>::Vector(Vector &&rhs)
+    : capacity_{rhs.capacity_ * delta}, data_{rhs.data_}, size_{rhs.size_} {
+  rhs.data_ = nullptr;
+  rhs.capacity_ = 0;
+  rhs.size_ = 0;
+}
+
+template <class T>
+Vector<T>::~Vector() {
+  if (data_ == nullptr) {
+    return;
+  }
+  delete[] data_;
+}
+
+template <class T>
+Vector<T> &Vector<T>::operator=(const Vector &arg) {
+  if (this == &arg) {
+    return *this;
+  }
+  delete[] data_;
+  size_ = arg.size_;
+  capacity_ = arg.capacity_;
+  data_ = new T[size_];
+  for (size_t i = 0; i < size_; ++i) {
+    data_[i] = arg.data_[i];
+  }
+  return *this;
+}
+template <class T>
+Vector<T> &Vector<T>::operator=(Vector &&rhs) {
+  data_ = rhs.data_;
+  capacity_ = rhs.capacity_;
+  size_ = rhs.size_;
+  rhs.data_ = nullptr;
+  rhs.capacity_ = 0;
+  rhs.size_ = 0;
+  return *this;
+}
+
+template <class T>
+T *Vector<T>::begin() const {
+  if (data_ == nullptr) {
+    return 0;
+  }
+  return data_;
+}
+
+template <class T>
+T *Vector<T>::end() const {
+  if (data_ == nullptr) {
+    return 0;
+  }
+  return data_ + size_;
+}
+
+template <class T>
+void Vector<T>::push_back(const T &value) {
+  if (size_ == capacity_) {
+    capacity_ *= delta;
+  }
+  data_[size_++] = value;
+}
+
+template <class T>
+void Vector<T>::push_back(T &&value) {
+  if (size_ == capacity_) {
+    capacity_ *= delta;
+  }
+  data_[size_++] = std::move(value);
+}
+
+template <class T>
+void Vector<T>::push_front(const T &value) {
+  if (data_ == nullptr) {
+    return;
+  }
+  insert(0, value);
+}
+
+template <class T>
+void Vector<T>::pop_back() {
+  if (data_ == nullptr) {
+    return;
+  }
+  --size_;
+}
+
+template <class T>
+template <class... Arg>
+void Vector<T>::emplace_back(Arg &&... args) {
+  new (&data_[size_++]) T{std::forward<Arg...>(args...)};
+}
+
+template <class T>
+T *Vector<T>::insert(const size_t pos, const T value) {
+  if (size_ < pos) {
+    return 0;
+  }
+  if (size_ == capacity_) {
+    capacity_ *= delta;
+  }
+  size_t pos_coun = pos;
```

Comment:
```suggestion
  if (size_ == capacity_) {
    capacity_ *= delta;
  }
  size_t pos_coun = pos;
```
вы не правильно используете подход перераспределения памяти. Нужно не капасити менять а саму память переалоцировать

Reply:
Done.